### DISCUSSION TITLE
When customInput is a input

### DISCUSSION THREAD

**Discussion by D-Thrane** [Hacker0x01/react-datepicker#3534] [Q&A]
I'm using customInput to render a custom input field that opens the date picker. When my focus is another input field in the form I need to click the datepicker input twice to open the date picker. I also noticed that every time I focus another input field in the form  the datepicker input re-renders.

Here is my code for the customInput:

```
  const DatePickerInput = forwardRef<HTMLInputElement, { value?: string; onClick?: () => void }>(
    ({ value, onClick }, ref) => (
      <>
        <input type="text" name={name} className="input__field cursor-pointer" onClick={onClick} ref={ref} value={value} title={tooltip || tooltip} readOnly={true} disabled={disabled} />
        <label className={`input__label text-faded ${required ? "label--required" : ""}`}>{placeholder}</label>
      </>
    ),
  );
```

**Comment by marcusstahlnacke** [comment]
Did you find a workaround?

**Reply by D-Thrane** [reply]
I hade to not use customInput and archive  what i needed with css

**Reply by marcusstahlnacke** [reply]
I solved it by wrapping the custominput with a useMemo

` const CustomInput = React.useMemo(() => {
    return forwardRef<ReactNode, any>(({ onClick }, ref) => (
      <FbDatepickerInput
        innerRef={ref}
        id={id + '__input'}
        label={label}
        disabled={disabled}
        required={required}
        validationFailedTooltip={validationFailedTooltip}
        tooltip={tooltipText}
        onClick={onClick}
      />
    ));
  }, []);`

**Reply by charliemourant** [reply]
@D-Thrane how did you modify the input field using css?
